testsuite/src/test/java/org/jboss/intersmash/tests/wildfly/web/cache/offload/infinispan/WildflyExternalizeSessionsToInfinispanApplication.java

```diff
@@ -0,0 +1,133 @@
+import org.jboss.intersmash.tests.wildfly.web.cache.offload.infinispan.util.InfinispanCommandLineBasedKeystoreGenerator;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Base64;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Application descriptor which represents an EAP application which will be provisioned via EAP image and that is
+ * configured in order to externalize distributed sessions to Infinispan/Red Hat Data Grid using only Infinispan
+ * subsystem.
+ */
+public class WildflyExternalizeSessionsToInfinispanApplication
+		implements WildflyImageOpenShiftApplication, WildflyApplicationConfiguration {
+	public static final String WILDFLY_APP_NAME = "wildfly";
+	private static final String TLS_CERTIFICATE_DIR_NAME = "/etc/secrets";
+
+	private final BuildInput buildInput;
+	private final List<EnvVar> environmentVariables;
+	private final List<Secret> secrets = new ArrayList<>();
+
+	public WildflyExternalizeSessionsToInfinispanApplication() throws IOException {
+		final String applicationDir = "wildfly/distributed-sessions-infinispan";
+		buildInput = new BuildInputBuilder()
+				.uri(IntersmashConfig.deploymentsRepositoryUrl())
+				.ref(IntersmashConfig.deploymentsRepositoryRef())
+				.build();
+
+		// setup environment variables
+		environmentVariables = new ArrayList<>();
+		environmentVariables.add(new EnvVarBuilder().withName("APP_NAME").withValue(Infinispan2ReplicasCustomCertificateService.INFINISPAN_APP_NAME).build());
+		environmentVariables
+				.add(new EnvVarBuilder().withName("INFINISPAN_HOST").withValue("$(INFINISPAN_SERVICE_HOST)").build());
+		environmentVariables
+				.add(new EnvVarBuilder().withName("INFINISPAN_PORT").withValue("$(INFINISPAN_SERVICE_PORT)").build());
+		// configure KUBE_PING, an invalidation-cache requires a functioning jgroups cluster.
+		environmentVariables.add(new EnvVarBuilder().withName("KUBERNETES_NAMESPACE")
+				.withValue(OpenShifts.master().getNamespace()).build());
+		//	credentials from Infinispan/Red Hat Data Grid APP custom secret
+		environmentVariables.add(new EnvVarBuilder().withName("CACHE_USERNAME")
+				.withValue(Infinispan2ReplicasCustomCertificateService.INFINISPAN_CUSTOM_CREDENTIALS_USERNAME).build());
+		environmentVariables.add(new EnvVarBuilder().withName("CACHE_PASSWORD")
+				.withValue(Infinispan2ReplicasCustomCertificateService.INFINISPAN_CUSTOM_CREDENTIALS_PASSWORD).build());
+		environmentVariables.add(new EnvVarBuilder().withName("TRUST_STORE_PATH")
+				.withValue(String.format("%s/%s", TLS_CERTIFICATE_DIR_NAME, Infinispan2ReplicasCustomCertificateService.TLS_CERTIFICATE_FILE_NAME)).build());
+
+		final String hostName = OpenShifts.master().generateHostname(Infinispan2ReplicasCustomCertificateService.INFINISPAN_APP_NAME);
+		final InfinispanCommandLineBasedKeystoreGenerator.InfinispanCertificate infinispanCertificate =
+				InfinispanCommandLineBasedKeystoreGenerator.generateInfinispanCertificate(
```

Comment:
Just use the existing utility provided by Intersmash, i.e. [CommandLineBasedKeystoreGenerator.java](https://github.com/Intersmash/intersmash/blob/main/core/src/main/java/org/jboss/intersmash/util/CommandLineBasedKeystoreGenerator.java)

Reply:
@fabiobrz see comment https://github.com/Intersmash/intersmash-tests/pull/37#discussion_r2405155498